agent_identity/src/document/aggregate.rs

```diff
@@ -374,65 +407,182 @@ impl Aggregate for Document {
                 Ok(vec![ServiceAdded { document_id, document }])
             }
             PublishDocument => {
-                // The API endpoint of an IOTA node, e.g. Hornet.
-                let api_endpoint = self
-                    .did_method
-                    .as_ref()
-                    .and_then(SupportedDidMethod::api_endpoint)
+                let document: IotaDocument = self.document.clone().ok_or(MissingDocumentError)?.into();
+
+                let did_method = self.did_method.ok_or(MissingDidMethodError)?;
+
+                // The API endpoint of an IOTA node
+                let api_endpoint = did_method
+                    .api_endpoint()
                     .ok_or_else(|| InvalidNodeEndpointError("missing `api_endpoint`".to_string()))?;
 
-                // Create a new client to interact with the IOTA ledger.
-                let iota_client: Client = Client::builder()
-                    .with_node(api_endpoint)
-                    .map_err(|_| InvalidNodeEndpointError(api_endpoint.to_string()))?
-                    .finish()
+                // Build a new IOTA client to interact with the IOTA ledger.
+                let mut iota_client_builder = IotaClientBuilder::default();
+
+                if let Some(iota_node_url) = config().iota_node_url.clone() {
```

Comment:
This seems to be duplicate code. Can this be extracted to a helper function?

Reply:
Added a helper function :+1: